src/modules/form-dyn-input-states.js

```diff
@@ -0,0 +1,78 @@
+(function ($) {
+
+	'use strict';
+	
+	var checkActiveAndErrorState = function (data) {
+		var field = data.field;
+		var fieldCtn = field.getCtn();
+		
+		if (fieldCtn.hasClass('js-input-radio')) {
+			fieldCtn = fieldCtn.add(fieldCtn.closest('.js-radio-list'));
+		}
+
+		window.updateState(fieldCtn, !!field.value(), 'active');
+		window.updateState(fieldCtn, !field.isValid(), 'error');
+	};
+
+	var onInputFocus = function (key, data) {
+		var field = data.field;
+		var fieldCtn = field.getCtn();
+
+		window.updateState(fieldCtn, true, 'active');
+	};
+
+	var onInputBlur = function (key, data) {
+		checkActiveAndErrorState(data);
+	};
+
+	var onInputChange = function (key, data) {
+		checkActiveAndErrorState(data);
+		
+		if (!!data.form.isValid()) {
+			App.mediator.notify('formDyn.isValid', {
+				form: data.form,
+				ctn: data.ctn
+			});
+		}
+	};
+
+	var onReset = function (key, data) {
+		var form = data.form;
+
+		form.eachFields(function (field) {
+			window.updateState(field.getCtn(), false, 'active');
```

Comment:
This pr needs to be merged when window.updateState will be implemented ? @nitriques

Reply:
Indeed. It should even be the new .fx thingy.

Reply:
Do we have an ETA for the 2.1 of the framework? I could definitely use it in a project next week.

Reply:
An that way I could update the form-dyn modules to work with .fx

Reply:
The problem is: it will take at least a day or two to patch all modules. But if you're willing to do it, I'll release this week.